In [1]:
import pandas as pd

In [2]:
transactions = pd.read_csv("20180519-transactions.csv", names=('date', 'number'))

Sources:
* https://blockchain.info/charts/n-transactions-per-block?showDataPoints=true

In [4]:
#plt = transactions.plot()

In [10]:
blocks_day = (60/10) * 24
blocks_day

144.0

As the bitcoin protocol tries to converge to one block per 10 minutes there are about 144 blocks per day

Sources:
* https://en.bitcoin.it/wiki/Protocol_rules

In [5]:
mean_transactions_block = transactions.number.mean()
print("There are about " + str(mean_transactions_block) + " transaction per Block")

There are about 1672.4470592245887 transaction per Block


In [14]:
transactions_day = blocks_day * mean_transactions_block
print("This results in about " + str(transactions_day) + " transactions per day")

This results in about 240832.37652834077 transactions per day


In [15]:
network_consumption_GW = 2.55
network_consumtpion_GWh_day = network_consumption_GW * 24
network_consumtpion_GWh_day

61.199999999999996

According to de Vries, the lower bound estimate for the power consumption of the whole network is 2.55 GW

Sources:
* A. de Vries, "Bitcoin's Growing Energy Problem", Joule , Volume 2 , Issue 5 , 801 - 805, https://www.cell.com/joule/fulltext/S2542-4351%2818%2930177-6

    
    


In [16]:
network_consumtpion_kWh_day = network_consumtpion_GWh_day * 1000 * 1000

In [17]:
consumption_transaction_kWh = network_consumtpion_kWh_day / transactions_day
consumption_transaction_kWh

254.1186566449801

## Validation

How much would the network consume, if revenues would just break even with electricity prices.   
We take the average US electricity price as a basis and the average earnings per day.

In [9]:
revenue = pd.read_csv("miners-revenue.csv", names=("date", "revenue"))
money_earned_per_day = revenue.revenue.mean()
money_earned_per_day

15778962.177637711

Sources:
* https://bitinfocharts.com/bitcoin/
* https://blockchain.info/de/charts/miners-revenue

In [79]:
electricity_prices = pd.read_csv("electricity_prices.csv", sep=";")
electricity_price_kWh = electricity_prices[electricity_prices.Region == "U.S. Total"]["All Sectors"].mean() / 100

Sources:
* https://www.eia.gov/electricity/monthly/epm_table_grapher.php?t=epmt_5_6_a

In [81]:
GWh_day = (money_earned_per_day / electricity_price_kWh) / 1000 / 1000
GWh_day

150.85049883018843

In [94]:
GWh_day * 1000 * 1000 / transactions_day

626.37134178027168

What the papers says per transaction:

In [18]:
print(300)

300
